### ISSUE TITLE
No players available to form syncgroup

### ISSUE THREAD

**Issue by chasut** [music-assistant/hass-music-assistant#3221]
### What version of Music Assistant has the issue?

2.4.0b7

### What version of the Home Assistant Integration have you got installed?

202411.4

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

Attempting to power on a group player (airplay) and receive the error "No players available to form syncgroup"

Setup is the same as when running 2.4.0b5 and works correctly (or at least as expected). Similar issue on 2.4.0b6 but did not have time to explore so just rolled back to previous version

Expected/previous behavior:
- Group player (airplay) with "Dynamic Group Members"
- Other players (airplay) are added through HA join action to allow for whole home audio management with HA 

### How to reproduce

Attempt to Power On the group player

### Music Providers

NA really, because I cannot even get the group player to power on, but I use Apple Music, Tune-in Radio, and the HA Radio Browser.

### Player Providers

Airplay

### Full log output

[MASlog.txt](https://github.com/user-attachments/files/17894441/MASlog.txt)

### What version of Home Assistant Core are your running

2024.11.3

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

macOS

**Comment by OzGav** [comment]
This will occur when trying to power on a sync group and the permanent members are not available. So what players have you added to that group and what is their status.

**Comment by chasut** [comment]
There is only one permanent member to this group.  It is available.  I tried to power the sync group on with the permanent member both in the powered on and powered off state with same results.  I restarted the MAS add-on with no change.  I tried with and without the "dynamic group members" with same results.

**Comment by neilflatley** [comment]
I saw the same in b6, I am using a squeezelite group - I reverted to b4 where this issue is not occurring

**Comment by Alundran** [comment]
Same issue for Slimproto groups, I can't power them on on the latest beta.

**Comment by jojo141185** [comment]
I had the same problem. Removing players from the group or replacing them with others, saving changes and adding them again solved the problem for me (but only temporarily). I use Snapcast clients and groups.

**Comment by OzGav** [comment]
The only time I see this problem is when players which are in the group are unavailable. You can tell by editing the config of the group and observing that some players have a weird name or MAC address. Is that the same for everyone reporting here?

**Comment by chasut** [comment]
Ok, finally found some time to explore a bit more.  Re-updated to 2.4.0b7.  Here were my steps and results.

Verified that the group player was setup correctly and that its sole member was available and working - same error
Deleted the group player and re-added as before (Dynamic group members with single member) - same error
Deleted and re-added the group player.  This time I didn't add a group member and it allowed to save anyway. - same error
Added back the single group member. - it WORKED! as before (mostly)

Thought maybe I was nuts so repeated steps to confirm.  Only issue is that group members were no longer reported in HA integration.  Remembered I had skipped that one too.  Updated the integration and now back in business.

Ok, last test.  Restart MAS and see if the settings hold.  - it did! but had the following weirdness.

- upon restart, the group player was powered on, as were the two of the individual players I used last, but they were NOT linked to the group player.
- I powered both off and was then allowed to link them to the group player.


Bottom line is I am now back in business on the latest beta.  But will need to manually adjust at restart each time.

**Comment by patienttruth** [comment]
I've been having this issue lately too (2.4.0b7). My sync group has two players in it, both from Sonos. 

I can play to either or both Sonos just fine, so they're not missing.

I restored a backup to 2.4.0b5 like ^ and now all is working. There wasn't much in the log even with player group on debug.

**Comment by OzGav** [comment]
This will get looked at soon

**Comment by vcdx71** [comment]
I'm seeing this in 2.3.4 as well, installed 2.4.0 B7 before seeing this issue and had it as well. Rolled back to 2.3.3 and all is good now.

**Comment by udochrist** [comment]
same here. 2.3.4 does not work with grouped homepods. rollback to 2.3.3 fixed the issue immediately. 
my groups are only created using the homepods/home.app, not in music assistant itself.

**Comment by OzGav** [comment]
Thanks. We don't need +1s. This is being looked at.

**Comment by udochrist** [comment]
Is there a fix for this coming to the stable release? I see it’s been merged to dev but there is no post-2.3.4 release implementing it.

**Comment by OzGav** [comment]
When the next stable release is done it will have this fix.
